truffle/src/main/ruby/core/rubinius/common/array.rb

```diff
@@ -564,4 +588,12 @@ def transpose
     out
   end
 
+  # from tuple.rb
+  def swap(a, b)
```

Comment:
Per my previous comment, this should just go into `truffle/src/main/ruby/core/rubinius/api/shims/tuple.rb`.